sync/rpc/server.go

```diff
@@ -308,3 +324,98 @@ func (*SyncAgentServer) BackupRestoreIncrementally(ctx context.Context, req *Bac
 	logrus.Infof("Done running %s %v", "sbackup", cmd.Args)
 	return &Empty{}, nil
 }
+
+// The APIs BackupAdd, BackupGet, Refresh, BackupDelete implement the CRUD interface for the backup object
+// The slice Backup.backupList is implemented similar to a FIFO queue.
+
+// BackupAdd creates a new backupList object and appends to the end of the list maintained by backup object
+func (b *Backup) BackupAdd(backupID string, backup *replica.Backup) error {
+	if backupID == "" {
+		return fmt.Errorf("empty backupID")
+	}
+
+	b.Lock()
+	b.backupList = append(b.backupList, &BackupInfo{
+		backupID:     backupID,
+		backupStatus: backup,
+	})
+	b.Unlock()
+
+	b.Refresh()
+	return nil
+}
+
+// BackupGet takes backupID input and will return the backup object corresponding to that backupID or error if not found
+func (b *Backup) BackupGet(backupID string) (*replica.Backup, error) {
+	if backupID == "" {
+		return nil, fmt.Errorf("empty backupID")
+	}
+
+	b.Refresh()
+
+	b.RLock()
+	defer b.RUnlock()
+
+	for _, info := range b.backupList {
+		if info.backupID == backupID {
+			return info.backupStatus, nil
+		}
+	}
+	return nil, fmt.Errorf("backup not found %v", backupID)
+}
+
+// remove deletes the object present at slice[index] and returns the remaining elements of slice yet maintaining
+// the original order of elements in the slice
+func (*Backup) remove(b []*BackupInfo, index int) []*BackupInfo {
+	if b == nil {
+		return nil
+	}
+	if index >= len(b) || index < 0 {
+		logrus.Errorf("BUG: index out of range")
```

Comment:
You're right. I missed error handling here. Yes, we need error for out of range operation though it would be treated as a bug. Make sure you provide enough information in the error message for that. This one doesn't have enough information.

Reply:
So should I change back the return value of Refresh function to error? Or should we just report it in logrus.Errorf?

Reply:
Change it back to error I meant. Log is not enough and ignore this will cause trouble down the line.